src/extractor/guidance/turn_handler.cpp

```diff
@@ -459,14 +522,62 @@ Intersection TurnHandler::assignRightTurns(const EdgeID via_edge,
     return intersection;
 }
 
-std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
-                                                          const Intersection &intersection) const
+// given two adjacent roads and `road1` being a candidate for a fork,
+// return false, if next road `road2` is also a fork candidate or
+// return true, if `road2` is not a suitable fork candidate and thus, `road1` the outermost fork
+bool isOutermostForkCandidate(const ConnectedRoad &road1, const ConnectedRoad &road2)
 {
+    const auto angle_between_next_road_and_straight = angularDeviation(road2.angle, STRAIGHT_ANGLE);
+    const auto angle_between_prev_road_and_next = angularDeviation(road1.angle, road2.angle);
+    const auto angle_between_prev_road_and_straight = angularDeviation(road1.angle, STRAIGHT_ANGLE);
+
+    // a road is a fork candidate if it is close to straight or
+    // close to a street that goes close to straight
+    // (reverse to find fork non-candidate)
+    if (angle_between_next_road_and_straight > NARROW_TURN_ANGLE)
+    {
+        if (angle_between_prev_road_and_next > NARROW_TURN_ANGLE ||
+            angle_between_prev_road_and_straight > GROUP_ANGLE)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+template <typename Iterator>
+std::size_t findOutermostForkCandidate(Iterator base, Iterator start, Iterator end)
+{
+    const auto outermost_fork_candidate = std::adjacent_find(start, end, isOutermostForkCandidate);
+    auto outermost_fork_candidate_index = std::distance(base, outermost_fork_candidate);
+    // if Iterator is a reverse iterator, the index is different
+    if (outermost_fork_candidate_index < 0)
+    {
+        outermost_fork_candidate_index = std::abs(outermost_fork_candidate_index) - 1;
+    }
```

Comment:
@daniel-j-h I have to search for the rightmost and the leftmost road and I use the same function but call it with the reverse iterator when looking for the rightmost one.

In that case (using the reverse iterator) the index will be negative and then I will have to do that. I added an explanation in the comment.